src/kernels/variables/kernelVariables.ts

```diff
@@ -80,7 +80,7 @@ export class KernelVariables implements IJupyterVariables {
         token?: CancellationToken
     ): Promise<IJupyterVariable | undefined> {
         // See if in the cache
-        const cache = this.notebookState.get(kernel.notebookDocument);
+        const cache = kernel.notebookDocument ? this.notebookState.get(kernel.notebookDocument) : undefined;
```

Comment:
Same here. Why does it need a notebookDocument?

Reply:
Fixed